### DISCUSSION TITLE
Connect Analytics Interface to external API such as Cloudflare Analytics

### DISCUSSION THREAD

**Discussion by reimertz** [umami-software/umami#684] [Ideas]
First, this solution is very beautiful and I love we are getting more solutions that compete with Google.

What I realized was that this solution has a hard dependency on client-side generated events to populate a hosted backend. 

I would have loved if there was a possibility to only use the (very beautiful and intuitive) interface and connect it to other sources such as Cloudflare Analytics. 

As it is currently architectured, would this be a slamdunk or a complete overhaul? If the latter, any directions on where I could start to add the functionality?

If not, maybe the best solution would be to fork this project and extract the UI elements and then connect that to Cloudflare.

**Comment by mikecao** [comment]
I don't know anything about Cloudflare but I doubt it would be an easy task. All the queries are made specifically for databases. You have to translate that somehow to a different API and the data format would have to match up as well.

**Reply by reimertz** [reply]
What is interesting is that Cloudflare Analytics have an [GraphQL API](https://developers.cloudflare.com/analytics/graphql-api). I might look into it. Thanks!
